numpy/_core/_type_aliases.py

```
@@ -76,18 +80,21 @@
 
 for k, v in _extra_aliases.items():
     sctypeDict[k] = allTypes[v]
+    del k, v
 
 # include extended precision sized aliases
 for is_complex, full_name in [(False, "longdouble"), (True, "clongdouble")]:
-    longdouble_type: type = allTypes[full_name]
+    longdouble_type = allTypes[full_name]
 
-    bits: int = dtype(longdouble_type).itemsize * 8
-    base_name: str = "complex" if is_complex else "float"
-    extended_prec_name: str = f"{base_name}{bits}"
+    bits = dtype(longdouble_type).itemsize * 8
+    base_name = "complex" if is_complex else "float"
+    extended_prec_name = f"{base_name}{bits}"
```

This probably looks very out of character, but on Python >=3.14 these were causing a module-level `__conditional_annotations__` name to be generated (no mention of this anywhere in the python docs). That was causing stubtest to report the following error:

```
numpy._core._type_aliases.__conditional_annotations__ is not present in stub
```

So it was indirectly also causing namespace pollution in that sense.

Usually I tend to ignore these kinds of stubtest errors, but in this case that'd be pretty annoying, seeing as it depends on the Python version. Ignoring errors in stubtest involves an "allowlist", and stubtest will also report an error for "unused allowlist entries". So to ignore this, we'd have to write a wrapper script that would conditionally select the allowlist, depending on the Python version used, and call stubtest with that.

And since these annotations were redundant anyways (type-checkers can infer those types themselves just fine), I thought we might as well get rid of them.

---

Thanks for an explanation! Yes, in that context it makes sense.